src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,198 @@
+package duke.command;
+
+import duke.task.TaskList;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+import duke.other.DateValidator;
+import duke.other.Storage;
+import duke.other.Ui;
+import duke.other.DukeException;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a command that adds on to the TaskList(i.e.Todo, Event, Deadline) that extends the Command class. An
+ * AddCommand object corresponds to a command type represented by a String followed by the necessary details of the
+ * command(e.g. Event Submit Assignment /at 2020/01/30 12:00).
+ */
+public class AddCommand extends Command {
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("yyyy/M/d");
+    private static final DateValidator DATE_VALIDATOR = new DateValidator(DATE_FORMATTER);
+
+    private static final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ofPattern("HH:mm");
+    private static final DateValidator TIME_VALIDATOR = new DateValidator(TIME_FORMATTER);
+    private String instruction;
+    private String details;
+
+    /**
+     * Constructs an AddCommand object.
+     *
+     * @param instruction Command type
+     * @param details     Details of command
+     */
+    public AddCommand(String instruction, String details) {
+        this.instruction = instruction;
+        this.details = details;
+    }
+
+    /**
+     * Reads the command type ("instruction") and executes the respective methods to handle the command.
+     *
+     * @param taskList Overall TaskList of all the Tasks
+     * @param ui       Overall Ui handling the ui of Duke
+     * @param storage  Storage handling the storage of the Tasks in TaskList
+     */
+    public void execute(TaskList taskList, Ui ui, Storage storage) {
+        switch (instruction) {
+        case "todo":
+            handleTodo(details, taskList);
+            break;
+        case "deadline":
+            handleDeadline(details, taskList);
+            break;
+        case "event":
+            handleEvent(details, taskList);
+            break;
+        default:
+            break;
+        }
+    }
```

Comment:
Since you have to still have to check and execute different methods to handle "todo", "deadline" and "event" commands, why not create separate classes for them? Same goes for your `ShowCommand` class